### ISSUE TITLE
Copying .ts files and renaming them confuses TypeScript

### ISSUE THREAD

**Issue by oliversalzburg** [microsoft/TypeScript#40616]
Issue Type: <b>Bug</b>

I encounter this in other circumstances, but the easiest way to reproduce this for me is in this small repro project I was building recently: https://github.com/oliversalzburg/mixin-mikro-orm

I would usually run a `tsc --watch` as a task in VSC, but in this project a `yarn build:watch` would achieve the same thing.

When I create copies of `.ts` files, like those in the `entities` folder, and then rename them (to create a similar entity), this will sooner or later trip up TypeScript in VSC. This manifests in an error in the problems panel:

> File '/tmp/mixin-mikro-orm/src/entities/Other copy 2.ts' not found.

The error points to `tsconfig.json` line 1.

The file reported as missing does expectedly not exist and I need to reload the TS server to recover from this.

This is not a recent regression. It has been happening for me for a long time.

VS Code version: Code 1.48.1 (3dd905126b34dcd4de81fa624eb3a8cbe7485f13, 2020-08-19T17:12:13.244Z)
OS version: Windows_NT x64 10.0.19041
Remote OS version: Linux x64 4.19.104-microsoft-standard
Remote OS version: Linux x64 4.19.104-microsoft-standard

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-6850K CPU @ 3.60GHz (12 x 3592)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|63.87GB (34.26GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|

|Item|Value|
|---|---|
|Remote|WSL: Ubuntu|
|OS|Linux x64 4.19.104-microsoft-standard|
|CPUs|Intel(R) Core(TM) i7-6850K CPU @ 3.60GHz (12 x 3591)|
|Memory (System)|50.11GB (35.04GB free)|
|VM|0%|

|Item|Value|
|---|---|
|Remote|WSL: Ubuntu|
|OS|Linux x64 4.19.104-microsoft-standard|
|CPUs|Intel(R) Core(TM) i7-6850K CPU @ 3.60GHz (12 x 3591)|
|Memory (System)|50.11GB (35.04GB free)|
|VM|0%|
</details><details><summary>Extensions (35)</summary>

Extension|Author (truncated)|Version
---|---|---
eslint-disable-snippets|drK|1.2.0
output-colorizer|IBM|0.1.2
vscode-docker|ms-|1.5.0
remote-containers|ms-|0.134.1
remote-ssh|ms-|0.51.0
remote-ssh-edit|ms-|0.51.0
remote-wsl|ms-|0.44.4
vscode-remote-extensionpack|ms-|0.20.0
vscode-icons|vsc|10.2.0
vscode-zipfs|arc|2.0.0
better-toml|bun|0.3.2
vscode-eslint|dba|2.1.8
xml|Dot|2.5.1
eslint-disable-snippets|drK|1.2.0
gitlens|eam|10.2.2
EditorConfig|Edi|0.15.1
prettier-vscode|esb|5.1.3
generateuuid|Fir|0.0.5
shell-format|fox|7.0.1
terraform|has|2.2.0
vscode-drawio|hed|0.7.2
vscode-esformatter|how|1.6.0
output-colorizer|IBM|0.1.2
code-beautifier|mic|2.3.3
vscode-lebab|mrm|2.6.0
vscode-puglint|mrm|2.3.0
vscode-docker|ms-|1.5.0
csharp|ms-|1.23.1
cpptools|ms-|0.29.0
hexeditor|ms-|1.2.1
js-debug-nightly|ms-|2020.8.1917
vscode-paste-image|mus|1.0.4
vscode-graphql|Pri|0.2.35
vscode-yaml|red|0.10.0
lit-plugin|run|1.2.1

(2 theme extensions excluded)

</details>

**Comment by sheetalkamat** [comment]
yarn install seems to be failing for me and then the issue does not repro.
Can you please make sure that the steps are reproducible and are issue with typescript@next?

```
:\temp\mixin-mikro-orm>yarn install
➤ YN0000: ┌ Resolution step
➤ YN0061: │ resolve-url@npm:0.2.1 is deprecated: https://github.com/lydell/resolve-url#deprecated
➤ YN0061: │ urix@npm:0.1.0 is deprecated: Please see https://github.com/lydell/urix#deprecated
➤ YN0000: └ Completed in 5.79s
➤ YN0000: ┌ Fetch step
➤ YN0013: │ to-regex@npm:3.0.2 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ xtend@npm:4.0.2 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ yallist@npm:4.0.0 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ type-check@npm:0.4.0 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ typescript@npm:4.1.3 can't be found in the cache and will be fetched from the remote registry
➤ YN0001: │ Error: typescript@patch:typescript@npm%3A4.1.3#builtin<compat/typescript>::version=4.1.3&hash=5b02a2: Cannot apply hunk #6
    at d (c:\temp\mixin-mikro-orm\.yarn\releases\yarn-berry.js:2:190674)
    at async c:\temp\mixin-mikro-orm\.yarn\releases\yarn-berry.js:2:189912
    at async h (c:\temp\mixin-mikro-orm\.yarn\releases\yarn-berry.js:2:188524)
    at async g (c:\temp\mixin-mikro-orm\.yarn\releases\yarn-berry.js:2:189885)
    at async q.fetchers.patchPackage (c:\temp\mixin-mikro-orm\.yarn\releases\yarn-berry.js:2:204417)
    at async B (c:\temp\mixin-mikro-orm\.yarn\releases\yarn-berry.js:2:226253)
    at async Q (c:\temp\mixin-mikro-orm\.yarn\releases\yarn-berry.js:2:226716)
    at async v (c:\temp\mixin-mikro-orm\.yarn\releases\yarn-berry.js:2:227192)
    at async C.fetchPackageFromCache (c:\temp\mixin-mikro-orm\.yarn\releases\yarn-berry.js:2:227431)
    at async q.fetchers.fetch (c:\temp\mixin-mikro-orm\.yarn\releases\yarn-berry.js:2:203286)
➤ YN0000: └ Completed in 15.16s
➤ YN0000: Failed with errors in 20.95s

```

After i run tsc even after failed install i cant repro the issue:
```
c:\temp\mixin-mikro-orm>node c:\TypeScript\built\local\tsc.js -watch --extendedDiagnostics
[3:00:04 PM] Starting compilation in watch mode...

Current directory: c:\temp\mixin-mikro-orm CaseSensitiveFileNames: false
FileWatcher:: Added:: WatchInfo: c:/temp/mixin-mikro-orm/tsconfig.json 2000 undefined Config file
Synchronizing program
CreatingProgramWith::
  roots: ["c:/temp/mixin-mikro-orm/src/index.ts","c:/temp/mixin-mikro-orm/src/mikro-orm.config.ts","c:/temp/mixin-mikro-orm/src/entities/Base.ts","c:/temp/mixin-mikro-orm/src/entities/Car.ts","c:/temp/mixin-mikro-orm/src/entities/ChangeTracked.ts"]
  options: {"allowSyntheticDefaultImports":true,"esModuleInterop":true,"emitDecoratorMetadata":true,"experimentalDecorators":true,"importHelpers":true,"module":1,"outDir":"c:/temp/mixin-mikro-orm/dist","sourceMap":true,"strict":true,"target":6,"incremental":true,"declaration":true,"watch":true,"extendedDiagnostics":true,"configFilePath":"c:/temp/mixin-mikro-orm/tsconfig.json"}
FileWatcher:: Added:: WatchInfo: c:/temp/mixin-mikro-orm/src/index.ts 250 undefined Source file
FileWatcher:: Added:: WatchInfo: c:/temp/mixin-mikro-orm/src/mikro-orm.config.ts 250 undefined Source file
FileWatcher:: Added:: WatchInfo: c:/temp/mixin-mikro-orm/src/entities/Base.ts 250 undefined Source file
FileWatcher:: Added:: WatchInfo: c:/temp/mixin-mikro-orm/src/entities/Car.ts 250 undefined Source file
FileWatcher:: Added:: WatchInfo: c:/temp/mixin-mikro-orm/src/entities/ChangeTracked.ts 250 undefined Source file
FileWatcher:: Added:: WatchInfo: c:/TypeScript/built/local/lib.es2019.full.d.ts 250 undefined Source file
FileWatcher:: Added:: WatchInfo: c:/TypeScript/built/local/lib.es2019.d.ts 250 undefined Source file
FileWatcher:: Added:: WatchInfo: c:/TypeScript/built/local/lib.es2018.d.ts 250 undefined Source file
FileWatcher:: Added:: WatchInfo: c:/TypeScript/built/local/lib.es2017.d.ts 250 undefined Source file
FileWatcher:: Added:: WatchInfo: c:/TypeScript/built/local/lib.es2016.d.ts 250 undefined Source file
FileWatcher:: Added:: WatchInfo: c:/TypeScript/built/local/lib.es2015.d.ts 250 undefined Source file
FileWatcher:: Added:: WatchInfo: c:/TypeScript/built/local/lib.es5.d.ts 250 undefined Source file
FileWatcher:: Added:: WatchInfo: c:/TypeScript/built/local/lib.es2015.core.d.ts 250 undefined Source file
FileWatcher:: Added:: WatchInfo: c:/TypeScript/built/local/lib.es2015.collection.d.ts 250 undefined Source file
FileWatcher:: Added:: WatchInfo: c:/TypeScript/built/local/lib.es2015.iterable.d.ts 250 undefined Source file
FileWatcher:: Added:: WatchInfo: c:/TypeScript/built/local/lib.es2015.symbol.d.ts 250 undefined Source file
FileWatcher:: Added:: WatchInfo: c:/TypeScript/built/local/lib.es2015.generator.d.ts 250 undefined Source file
FileWatcher:: Added:: WatchInfo: c:/TypeScript/built/local/lib.es2015.promise.d.ts 250 undefined Source file
FileWatcher:: Added:: WatchInfo: c:/TypeScript/built/local/lib.es2015.proxy.d.ts 250 undefined Source file
FileWatcher:: Added:: WatchInfo: c:/TypeScript/built/local/lib.es2015.reflect.d.ts 250 undefined Source file
FileWatcher:: Added:: WatchInfo: c:/TypeScript/built/local/lib.es2015.symbol.wellknown.d.ts 250 undefined Source file
FileWatcher:: Added:: WatchInfo: c:/TypeScript/built/local/lib.es2016.array.include.d.ts 250 undefined Source file
FileWatcher:: Added:: WatchInfo: c:/TypeScript/built/local/lib.es2017.object.d.ts 250 undefined Source file
FileWatcher:: Added:: WatchInfo: c:/TypeScript/built/local/lib.es2017.sharedmemory.d.ts 250 undefined Source file
FileWatcher:: Added:: WatchInfo: c:/TypeScript/built/local/lib.es2017.string.d.ts 250 undefined Source file
FileWatcher:: Added:: WatchInfo: c:/TypeScript/built/local/lib.es2017.intl.d.ts 250 undefined Source file
FileWatcher:: Added:: WatchInfo: c:/TypeScript/built/local/lib.es2017.typedarrays.d.ts 250 undefined Source file
FileWatcher:: Added:: WatchInfo: c:/TypeScript/built/local/lib.es2018.asynciterable.d.ts 250 undefined Source file
FileWatcher:: Added:: WatchInfo: c:/TypeScript/built/local/lib.es2018.asyncgenerator.d.ts 250 undefined Source file
FileWatcher:: Added:: WatchInfo: c:/TypeScript/built/local/lib.es2018.promise.d.ts 250 undefined Source file
FileWatcher:: Added:: WatchInfo: c:/TypeScript/built/local/lib.es2018.regexp.d.ts 250 undefined Source file
FileWatcher:: Added:: WatchInfo: c:/TypeScript/built/local/lib.es2018.intl.d.ts 250 undefined Source file
FileWatcher:: Added:: WatchInfo: c:/TypeScript/built/local/lib.es2019.array.d.ts 250 undefined Source file
FileWatcher:: Added:: WatchInfo: c:/TypeScript/built/local/lib.es2019.object.d.ts 250 undefined Source file
FileWatcher:: Added:: WatchInfo: c:/TypeScript/built/local/lib.es2019.string.d.ts 250 undefined Source file
FileWatcher:: Added:: WatchInfo: c:/TypeScript/built/local/lib.es2019.symbol.d.ts 250 undefined Source file
FileWatcher:: Added:: WatchInfo: c:/TypeScript/built/local/lib.dom.d.ts 250 undefined Source file
FileWatcher:: Added:: WatchInfo: c:/TypeScript/built/local/lib.webworker.importscripts.d.ts 250 undefined Source file
FileWatcher:: Added:: WatchInfo: c:/TypeScript/built/local/lib.scripthost.d.ts 250 undefined Source file
FileWatcher:: Added:: WatchInfo: c:/TypeScript/built/local/lib.dom.iterable.d.ts 250 undefined Source file
DirectoryWatcher:: Added:: WatchInfo: c:/temp/mixin-mikro-orm/src 1 undefined Failed Lookup Locations
Elapsed:: 1.2226998805999756ms DirectoryWatcher:: Added:: WatchInfo: c:/temp/mixin-mikro-orm/src 1 undefined Failed Lookup Locations
DirectoryWatcher:: Added:: WatchInfo: c:/temp/mixin-mikro-orm/node_modules 1 undefined Failed Lookup Locations
Elapsed:: 0.2822000980377197ms DirectoryWatcher:: Added:: WatchInfo: c:/temp/mixin-mikro-orm/node_modules 1 undefined Failed Lookup Locations
DirectoryWatcher:: Added:: WatchInfo: c:/temp/mixin-mikro-orm/node_modules/@types 1 undefined Type roots
Elapsed:: 0.4218001365661621ms DirectoryWatcher:: Added:: WatchInfo: c:/temp/mixin-mikro-orm/node_modules/@types 1 undefined Type roots
src/entities/Base.ts:1:36 - error TS2307: Cannot find module '@mikro-orm/core' or its corresponding type declarations.

1 import { Entity, PrimaryKey } from "@mikro-orm/core";
                                     ~~~~~~~~~~~~~~~~~

src/entities/Base.ts:3:1 - error TS2354: This syntax requires an imported helper but module 'tslib' cannot be found.

3 @Entity()
  ~~~~~~~~~

src/entities/Car.ts:1:34 - error TS2307: Cannot find module '@mikro-orm/core' or its corresponding type declarations.

1 import { Entity, Property } from "@mikro-orm/core";
                                   ~~~~~~~~~~~~~~~~~

src/entities/ChangeTracked.ts:1:26 - error TS2307: Cannot find module '@mikro-orm/core' or its corresponding type declarations.

1 import { Property } from "@mikro-orm/core";
                           ~~~~~~~~~~~~~~~~~

src/index.ts:1:26 - error TS2307: Cannot find module '@mikro-orm/core' or its corresponding type declarations.

1 import { MikroORM } from "@mikro-orm/core";
                           ~~~~~~~~~~~~~~~~~

src/mikro-orm.config.ts:1:33 - error TS2307: Cannot find module '@mikro-orm/core' or its corresponding type declarations.

1 import { MikroORMOptions } from "@mikro-orm/core";
                                  ~~~~~~~~~~~~~~~~~

src/mikro-orm.config.ts:2:41 - error TS2307: Cannot find module '@mikro-orm/reflection' or its corresponding type declarations.

2 import { TsMorphMetadataProvider } from "@mikro-orm/reflection";
                                          ~~~~~~~~~~~~~~~~~~~~~~~

src/mikro-orm.config.ts:8:11 - error TS2304: Cannot find name '__dirname'.

8  baseDir: __dirname + "/..",
            ~~~~~~~~~

[3:00:05 PM] Found 8 errors. Watching for file changes.

Files:                         40
Lines of Library:           28304
Lines of Definitions:           0
Lines of TypeScript:           69
Lines of JavaScript:            0
Lines of JSON:                  0
Lines of Other:                 0
Nodes of Library:          122226
Nodes of Definitions:           0
Nodes of TypeScript:          244
Nodes of JavaScript:            0
Nodes of JSON:                  0
Nodes of Other:                 0
Identifiers:                44893
Symbols:                    26762
Types:                         78
Instantiations:                 0
Memory used:               60450K
Assignability cache size:       0
Identity cache size:            0
Subtype cache size:             0
Strict subtype cache size:      0
I/O Read time:              0.02s
Parse time:                 0.64s
ResolveModule time:         0.02s
Program time:               0.73s
Bind time:                  0.26s
Total time:                 0.99s
DirectoryWatcher:: Added:: WatchInfo: c:/temp/mixin-mikro-orm/src 1 undefined Wild card directory
Elapsed:: 0.838900089263916ms DirectoryWatcher:: Added:: WatchInfo: c:/temp/mixin-mikro-orm/src 1 undefined Wild card directory
DirectoryWatcher:: Triggered with c:/temp/mixin-mikro-orm/src/entities/Car - Copy.ts :: WatchInfo: c:/temp/mixin-mikro-orm/src 1 undefined Failed Lookup Locations
Scheduling invalidateFailedLookup
Elapsed:: 5.810899972915649ms DirectoryWatcher:: Triggered with c:/temp/mixin-mikro-orm/src/entities/Car - Copy.ts :: WatchInfo: c:/temp/mixin-mikro-orm/src 1 undefined Failed Lookup Locations
DirectoryWatcher:: Triggered with c:/temp/mixin-mikro-orm/src/entities/Car - Copy.ts :: WatchInfo: c:/temp/mixin-mikro-orm/src 1 undefined Wild card directory
Scheduling update
Elapsed:: 1.0614001750946045ms DirectoryWatcher:: Triggered with c:/temp/mixin-mikro-orm/src/entities/Car - Copy.ts :: WatchInfo: c:/temp/mixin-mikro-orm/src 1 undefined Wild card directory
[3:01:12 PM] File change detected. Starting incremental compilation...

Reloading new file names and options
Synchronizing program
CreatingProgramWith::
  roots: ["c:/temp/mixin-mikro-orm/src/index.ts","c:/temp/mixin-mikro-orm/src/mikro-orm.config.ts","c:/temp/mixin-mikro-orm/src/entities/Base.ts","c:/temp/mixin-mikro-orm/src/entities/Car - Copy.ts","c:/temp/mixin-mikro-orm/src/entities/Car.ts","c:/temp/mixin-mikro-orm/src/entities/ChangeTracked.ts"]
  options: {"allowSyntheticDefaultImports":true,"esModuleInterop":true,"emitDecoratorMetadata":true,"experimentalDecorators":true,"importHelpers":true,"module":1,"outDir":"c:/temp/mixin-mikro-orm/dist","sourceMap":true,"strict":true,"target":6,"incremental":true,"declaration":true,"watch":true,"extendedDiagnostics":true,"configFilePath":"c:/temp/mixin-mikro-orm/tsconfig.json"}
FileWatcher:: Added:: WatchInfo: c:/temp/mixin-mikro-orm/src/entities/Car - Copy.ts 250 undefined Source file
src/entities/Base.ts:1:36 - error TS2307: Cannot find module '@mikro-orm/core' or its corresponding type declarations.

1 import { Entity, PrimaryKey } from "@mikro-orm/core";
                                     ~~~~~~~~~~~~~~~~~

src/entities/Base.ts:3:1 - error TS2354: This syntax requires an imported helper but module 'tslib' cannot be found.

3 @Entity()
  ~~~~~~~~~

src/entities/Car - Copy.ts:1:34 - error TS2307: Cannot find module '@mikro-orm/core' or its corresponding type declarations.

1 import { Entity, Property } from "@mikro-orm/core";
                                   ~~~~~~~~~~~~~~~~~

src/entities/Car - Copy.ts:5:1 - error TS2354: This syntax requires an imported helper but module 'tslib' cannot be found.

5 @Entity()
  ~~~~~~~~~

src/entities/Car.ts:1:34 - error TS2307: Cannot find module '@mikro-orm/core' or its corresponding type declarations.

1 import { Entity, Property } from "@mikro-orm/core";
                                   ~~~~~~~~~~~~~~~~~

src/entities/ChangeTracked.ts:1:26 - error TS2307: Cannot find module '@mikro-orm/core' or its corresponding type declarations.

1 import { Property } from "@mikro-orm/core";
                           ~~~~~~~~~~~~~~~~~

src/index.ts:1:26 - error TS2307: Cannot find module '@mikro-orm/core' or its corresponding type declarations.

1 import { MikroORM } from "@mikro-orm/core";
                           ~~~~~~~~~~~~~~~~~

src/mikro-orm.config.ts:1:33 - error TS2307: Cannot find module '@mikro-orm/core' or its corresponding type declarations.

1 import { MikroORMOptions } from "@mikro-orm/core";
                                  ~~~~~~~~~~~~~~~~~

src/mikro-orm.config.ts:2:41 - error TS2307: Cannot find module '@mikro-orm/reflection' or its corresponding type declarations.

2 import { TsMorphMetadataProvider } from "@mikro-orm/reflection";
                                          ~~~~~~~~~~~~~~~~~~~~~~~

src/mikro-orm.config.ts:8:11 - error TS2304: Cannot find name '__dirname'.

8  baseDir: __dirname + "/..",
            ~~~~~~~~~

[3:01:12 PM] Found 10 errors. Watching for file changes.

Files:                         41
Lines of Library:           28304
Lines of Definitions:           0
Lines of TypeScript:           79
Lines of JavaScript:            0
Lines of JSON:                  0
Lines of Other:                 0
Nodes of Library:          122226
Nodes of Definitions:           0
Nodes of TypeScript:          285
Nodes of JavaScript:            0
Nodes of JSON:                  0
Nodes of Other:                 0
Identifiers:                44904
Symbols:                    26816
Types:                        125
Instantiations:                20
Memory used:               57710K
Assignability cache size:       6
Identity cache size:            0
Subtype cache size:             0
Strict subtype cache size:      0
ResolveModule time:         0.00s
I/O Read time:              0.00s
Parse time:                 0.00s
Program time:               0.02s
Bind time:                  0.00s
Check time:                 0.04s
transformTime time:         0.04s
commentTime time:           0.01s
printTime time:             0.07s
Emit time:                  0.08s
Source Map time:            0.00s
I/O Write time:             0.00s
Total time:                 0.13s
DirectoryWatcher:: Triggered with c:/temp/mixin-mikro-orm/src/entities/Car - Copy.ts :: WatchInfo: c:/temp/mixin-mikro-orm/src 1 undefined Failed Lookup Locations
Scheduling invalidateFailedLookup
Elapsed:: 5.893900156021118ms DirectoryWatcher:: Triggered with c:/temp/mixin-mikro-orm/src/entities/Car - Copy.ts :: WatchInfo: c:/temp/mixin-mikro-orm/src 1 undefined Failed Lookup Locations
DirectoryWatcher:: Triggered with c:/temp/mixin-mikro-orm/src/entities/Car2.ts :: WatchInfo: c:/temp/mixin-mikro-orm/src 1 undefined Failed Lookup Locations
Scheduling invalidateFailedLookup, Cancelled earlier one
Elapsed:: 1.2525999546051025ms DirectoryWatcher:: Triggered with c:/temp/mixin-mikro-orm/src/entities/Car2.ts :: WatchInfo: c:/temp/mixin-mikro-orm/src 1 undefined Failed Lookup Locations
DirectoryWatcher:: Triggered with c:/temp/mixin-mikro-orm/src/entities/Car - Copy.ts :: WatchInfo: c:/temp/mixin-mikro-orm/src 1 undefined Wild card directory
Scheduling update
Elapsed:: 0.8756999969482422ms DirectoryWatcher:: Triggered with c:/temp/mixin-mikro-orm/src/entities/Car - Copy.ts :: WatchInfo: c:/temp/mixin-mikro-orm/src 1 undefined Wild card directory
DirectoryWatcher:: Triggered with c:/temp/mixin-mikro-orm/src/entities/Car2.ts :: WatchInfo: c:/temp/mixin-mikro-orm/src 1 undefined Wild card directory
Scheduling update
Elapsed:: 0.8069000244140625ms DirectoryWatcher:: Triggered with c:/temp/mixin-mikro-orm/src/entities/Car2.ts :: WatchInfo: c:/temp/mixin-mikro-orm/src 1 undefined Wild card directory
FileWatcher:: Triggered with c:/temp/mixin-mikro-orm/src/entities/Car - Copy.ts 2:: WatchInfo: c:/temp/mixin-mikro-orm/src/entities/Car - Copy.ts 250 undefined Source file
Scheduling update
Elapsed:: 9.690600156784058ms FileWatcher:: Triggered with c:/temp/mixin-mikro-orm/src/entities/Car - Copy.ts 2:: WatchInfo: c:/temp/mixin-mikro-orm/src/entities/Car - Copy.ts 250 undefined Source file
[3:01:57 PM] File change detected. Starting incremental compilation...

Reloading new file names and options
Synchronizing program
CreatingProgramWith::
  roots: ["c:/temp/mixin-mikro-orm/src/index.ts","c:/temp/mixin-mikro-orm/src/mikro-orm.config.ts","c:/temp/mixin-mikro-orm/src/entities/Base.ts","c:/temp/mixin-mikro-orm/src/entities/Car.ts","c:/temp/mixin-mikro-orm/src/entities/Car2.ts","c:/temp/mixin-mikro-orm/src/entities/ChangeTracked.ts"]
  options: {"allowSyntheticDefaultImports":true,"esModuleInterop":true,"emitDecoratorMetadata":true,"experimentalDecorators":true,"importHelpers":true,"module":1,"outDir":"c:/temp/mixin-mikro-orm/dist","sourceMap":true,"strict":true,"target":6,"incremental":true,"declaration":true,"watch":true,"extendedDiagnostics":true,"configFilePath":"c:/temp/mixin-mikro-orm/tsconfig.json"}
FileWatcher:: Added:: WatchInfo: c:/temp/mixin-mikro-orm/src/entities/Car2.ts 250 undefined Source file
FileWatcher:: Close:: WatchInfo: c:/temp/mixin-mikro-orm/src/entities/Car - Copy.ts 250 undefined Source file
src/entities/Base.ts:1:36 - error TS2307: Cannot find module '@mikro-orm/core' or its corresponding type declarations.

1 import { Entity, PrimaryKey } from "@mikro-orm/core";
                                     ~~~~~~~~~~~~~~~~~

src/entities/Base.ts:3:1 - error TS2354: This syntax requires an imported helper but module 'tslib' cannot be found.

3 @Entity()
  ~~~~~~~~~

src/entities/Car.ts:1:34 - error TS2307: Cannot find module '@mikro-orm/core' or its corresponding type declarations.

1 import { Entity, Property } from "@mikro-orm/core";
                                   ~~~~~~~~~~~~~~~~~

src/entities/Car2.ts:1:34 - error TS2307: Cannot find module '@mikro-orm/core' or its corresponding type declarations.

1 import { Entity, Property } from "@mikro-orm/core";
                                   ~~~~~~~~~~~~~~~~~

src/entities/Car2.ts:5:1 - error TS2354: This syntax requires an imported helper but module 'tslib' cannot be found.

5 @Entity()
  ~~~~~~~~~

src/entities/ChangeTracked.ts:1:26 - error TS2307: Cannot find module '@mikro-orm/core' or its corresponding type declarations.

1 import { Property } from "@mikro-orm/core";
                           ~~~~~~~~~~~~~~~~~

src/index.ts:1:26 - error TS2307: Cannot find module '@mikro-orm/core' or its corresponding type declarations.

1 import { MikroORM } from "@mikro-orm/core";
                           ~~~~~~~~~~~~~~~~~

src/mikro-orm.config.ts:1:33 - error TS2307: Cannot find module '@mikro-orm/core' or its corresponding type declarations.

1 import { MikroORMOptions } from "@mikro-orm/core";
                                  ~~~~~~~~~~~~~~~~~

src/mikro-orm.config.ts:2:41 - error TS2307: Cannot find module '@mikro-orm/reflection' or its corresponding type declarations.

2 import { TsMorphMetadataProvider } from "@mikro-orm/reflection";
                                          ~~~~~~~~~~~~~~~~~~~~~~~

src/mikro-orm.config.ts:8:11 - error TS2304: Cannot find name '__dirname'.

8  baseDir: __dirname + "/..",
            ~~~~~~~~~

[3:01:57 PM] Found 10 errors. Watching for file changes.

Files:                         41
Lines of Library:           28304
Lines of Definitions:           0
Lines of TypeScript:           79
Lines of JavaScript:            0
Lines of JSON:                  0
Lines of Other:                 0
Nodes of Library:          122226
Nodes of Definitions:           0
Nodes of TypeScript:          285
Nodes of JavaScript:            0
Nodes of JSON:                  0
Nodes of Other:                 0
Identifiers:                44904
Symbols:                    26816
Types:                        125
Instantiations:                20
Memory used:               59406K
Assignability cache size:       6
Identity cache size:            0
Subtype cache size:             0
Strict subtype cache size:      0
ResolveModule time:         0.01s
I/O Read time:              0.00s
Parse time:                 0.00s
Program time:               0.03s
Bind time:                  0.00s
Check time:                 0.01s
transformTime time:         0.01s
commentTime time:           0.01s
printTime time:             0.04s
Emit time:                  0.04s
Source Map time:            0.00s
I/O Write time:             0.00s
Total time:                 0.08s
```

**Comment by oliversalzburg** [comment]
I managed to fix the `install` and `build` part, but the project also doesn't start for me as it used to. There have probably been some unexpected changes in the dependencies.

However, receiving the initial error, which indicates to me that a patch to typescript is applied by yarn 2, suggests that this is likely a bug in that patch and not in TypeScript itself.

When I opened the issue, I wasn't fully aware of how all these components interact. Without being able to isolate this to an unpatched TypeScript, it shouldn't be handled here.
